packages/gensx-core/tests/restore-checkpoint.test.ts

```diff
@@ -0,0 +1,176 @@
+    const mockOnRestoreCheckpoint = vi.fn().mockResolvedValue(undefined);
+    const workflowContext = createWorkflowContext({
+      onRestoreCheckpoint: mockOnRestoreCheckpoint,
+    });
+    const executionContext = new ExecutionContext({});
+    const contextWithWorkflow = executionContext.withContext({
+      [WORKFLOW_CONTEXT_SYMBOL]: workflowContext,
+    });
+
+    // Mock console.error to capture the expected log
+    const consoleErrorSpy = vi
+      .spyOn(console, "error")
+      .mockImplementation(() => {
+        // Mock implementation
+      });
+
+    await withContext(contextWithWorkflow, async () => {
+      // First create a checkpoint
+      createCheckpoint({ label: "test-restore" });
+
+      // Mock the waitForPendingUpdates method
+      vi.spyOn(
+        workflowContext.checkpointManager,
+        "waitForPendingUpdates",
+      ).mockResolvedValue();
+
+      // Now restore it - should complete without throwing
+      const feedback = { message: "test feedback" };
+      await restoreCheckpoint("test-restore", feedback);
+
+      // Verify the callback was called with correct parameters
+      expect(mockOnRestoreCheckpoint).toHaveBeenCalledWith(
+        workflowContext.checkpointLabelMap.get("test-restore"),
+        feedback,
+      );
+
+      // Verify error was logged
+      expect(consoleErrorSpy).toHaveBeenCalledWith(
+        "[GenSX] Restoring checkpoints is not supported in this environment.",
+      );
+    });
+
+    consoleErrorSpy.mockRestore();
+  });
+
+  test("checkpoint restore function calls onRestoreCheckpoint", async () => {
+    const mockOnRestoreCheckpoint = vi.fn().mockResolvedValue(undefined);
+    const workflowContext = createWorkflowContext({
+      onRestoreCheckpoint: mockOnRestoreCheckpoint,
+    });
+    const executionContext = new ExecutionContext({});
+    const contextWithWorkflow = executionContext.withContext({
+      [WORKFLOW_CONTEXT_SYMBOL]: workflowContext,
+    });
+
+    const consoleErrorSpy = vi
+      .spyOn(console, "error")
+      .mockImplementation(() => {
+        // Mock implementation
+      });
+
+    await withContext(contextWithWorkflow, async () => {
+      const checkpoint = createCheckpoint({ label: "test-direct-restore" });
+
+      // Mock the waitForPendingUpdates method
+      vi.spyOn(
+        workflowContext.checkpointManager,
+        "waitForPendingUpdates",
+      ).mockResolvedValue();
+
+      const feedback = { value: 42 };
+      await checkpoint.restore(feedback);
+
+      expect(mockOnRestoreCheckpoint).toHaveBeenCalledWith(
+        workflowContext.checkpointLabelMap.get("test-direct-restore"),
+        feedback,
+      );
+
+      expect(consoleErrorSpy).toHaveBeenCalledWith(
+        "[GenSX] Restoring checkpoints is not supported in this environment.",
+      );
+    });
+
+    consoleErrorSpy.mockRestore();
+  });
+
+  test("createCheckpoint respects maxRestores option", () => {
```

Comment:
Feels like we should have a test that ensures that a workflow can create multiple checkpoints and restore from each of them.